### ISSUE TITLE
Feature Request: Typst Support

### ISSUE THREAD

**Issue by farinc** [bartosz-antosik/vscode-spellright#570]
I think the addition of [Typst](https://typst.app/) support would be great, it being a sort of successor to LaTex with already great support in VS Code. It is also very similar to Markdown.

**Comment by memeplex** [comment]
I was going to ask the same.

I think the only thing to do is to support the "text" class in `parserByClass`: 

<img width="293" alt="image" src="https://github.com/bartosz-antosik/vscode-spellright/assets/2845433/9fb95257-8e6d-4e3e-b5e6-23e808ca305b">

**Comment by memeplex** [comment]
No, I assumed you were using vscode syntax classes, but I see there are custom parsers for each supported value.

Isn't there more performant filters based on syntax scopes so as not to use regexs nor implement yet another parser by hand?
